modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/instance/test/AddDefaultAssetVocabulariesInitialRequestPortalInstanceLifecycleListenerTest.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.content.dashboard.web.internal.instance.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.asset.kernel.model.AssetCategoryConstants;
+import com.liferay.asset.kernel.model.AssetVocabulary;
+import com.liferay.asset.kernel.service.AssetVocabularyLocalService;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.SetUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portlet.asset.util.AssetVocabularySettingsHelper;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Attila Bakay
+ */
+@RunWith(Arquillian.class)
+public class
+	AddDefaultAssetVocabulariesInitialRequestPortalInstanceLifecycleListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testPortalInstanceRegistered() throws Exception {
+		Company company = _companyLocalService.getCompany(
+			TestPropsValues.getCompanyId());
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.fetchGroupVocabulary(
+				company.getGroupId(), "audience");
+
+		long[] originalClassNameIdClassNameIds =
+			assetVocabulary.getSelectedClassNameIds();
+
+		long classNameId = originalClassNameIdClassNameIds[0];
+
+		Set<Long> classNameIds = SetUtil.fromArray(
+			originalClassNameIdClassNameIds);
+
+		classNameIds.remove(classNameId);
+
+		_updateVocabulary(assetVocabulary, ArrayUtil.toLongArray(classNameIds));
+
+		_portalInstanceLifecycleListener.portalInstanceRegistered(company);
+
+		assetVocabulary = _assetVocabularyLocalService.fetchGroupVocabulary(
+			company.getGroupId(), "audience");
+
+		Assert.assertFalse(
+			ArrayUtil.contains(
+				assetVocabulary.getSelectedClassNameIds(), classNameId));
+
+		_updateVocabulary(assetVocabulary, originalClassNameIdClassNameIds);
```

Comment:
After the first `_updateVocabulary` I woud put everything in a `try` block and put the second `_updateVocabulary` in the `finally` block. If something bad happens or the assert fails the second update will not be done...

Reply:
done